rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -185,6 +188,88 @@ __lockless_has_parameter(
   return parameters.find(name) != parameters.end();
 }
 
+// see https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon
+#define ULP 100
+RCLCPP_LOCAL
+bool
+__are_doubles_equal(double x, double y, int ulp = ULP)
+{
+  return std::abs(x - y) <= std::numeric_limits<double>::epsilon() * std::abs(x + y) * ulp;
+}
+
+RCLCPP_LOCAL
+bool
+__check_parameter_value_in_range(
+  const rcl_interfaces::msg::ParameterDescriptor & descriptor,
+  const rclcpp::ParameterValue & value)
+{
+  if (!descriptor.integer_range.empty() && value.get_type() == rclcpp::PARAMETER_INTEGER) {
+    int64_t v = value.get<int64_t>();
+    auto integer_range = descriptor.integer_range.at(0);
+    if ((v < integer_range.from_value) || (v > integer_range.to_value)) {
+      return false;
+    }
+    if (v == integer_range.to_value) {
+      return true;
+    }
+    if (integer_range.step == 0) {
+      return true;
+    }
+    if (((v - integer_range.from_value) % integer_range.step) == 0) {
+      return true;
+    }
+    return false;
+  }
+
+  if (!descriptor.floating_point_range.empty() && value.get_type() == rclcpp::PARAMETER_DOUBLE) {
+    double v = value.get<double>();
+    auto fp_range = descriptor.floating_point_range.at(0);
+    if (__are_doubles_equal(v, fp_range.from_value) ||
+      __are_doubles_equal(v, fp_range.to_value))
+    {
```

Comment:
nitpick: move `{` to the line above.

Reply:
It's not recommended when the condition of the if is multi-line, because L228 and L230 will have the same indentation. Actually, uncrustify will complain.